assemble.py

```diff
@@ -47,20 +47,26 @@ def losetup(path: str) -> str:
     # up, thus the "No such file or directory". When this happens, we need to
     # remount /dev in order to see the new device.
     try:
+        # get the next available loop device
+        loop_device = cmd('losetup', '-f', capture=True).decode().rstrip()
         cmd('losetup', '-P', '-f', path)
     except subprocess.CalledProcessError:
         logger.error('losetup bug found, remounting /dev to work around')
         remount_dev()
+        # get the next available loop device
+        loop_device = cmd('losetup', '-f', capture=True).decode().rstrip()
         cmd('losetup', '-P', '-f', path)
 
     # The -P in losetup scans for partitions and will create entries like:
     # /dev/loopXp1. Since these are new /dev entries, we have to remount /dev
     remount_dev()
 
+    # make sure that the most recently created loop device represents a given system image (`path`)
```

Comment:
I'm assuming you hit this in local development and its not really something that can happen in actual CI, right?

Reply:
Yes, I hit it locally. I think it may happen in actual CI though. E.G. if a run is cancelled or a run container is killed, so `losetup -d` is never called. As result, the next run on that host/worker can use a loop device that mounts the previous run's image.